File: Intersection of two Integer arrays/arrayintersection.java
//by Siddharth Saurabh (https://github.com/siddhartthecoder)
   import java.util.*;

   public class arrayintersection {

    public static void main(String[] args) {

     Scanner sc = new Scanner(System.in);

     System.out.println("enter number of elements in first array");

     int n = sc.nextInt();

     int arr1[] = new int[n];

     System.out.println("enter elements");

     for (int i = 0; i < n; i++) { //for reading array 1
      arr1[i] = sc.nextInt();

     }

     System.out.println("enter number of elements in second array");

     int m = sc.nextInt();

     int arr2[] = new int[m];

     System.out.println("enter elements");

     for (int i = 0; i < m; i++) { //for reading array 2
      arr2[i] = sc.nextInt();

     }
     System.out.println("The intersection is :");
     //Intersection logic
     for (int i = 0; i < arr1.length; i++) {
      for (int j = 0; j < arr2.length; j++) {
       if (arr1[i] == arr2[j]) {
        System.out.print(" " + arr2[j] + " ");
       }
      }
     }
     System.out.println("");

    }

   }

File: Intersection of two Integer arrays/arrayIntersection.c
#include <stdio.h>

int main()
{
    int sa, sb;
    printf("Inform the size of the first list: ");
    scanf("%d", &sa);
    int ma[sa];
    printf("Inform the size of the second list: ");
    scanf("%d", &sb);
    int mb[sb];
    printf("Inform the list of integers space separated:\n");
    for (int i = 0; i < sa; i++) {
        scanf("%d", &ma[i]);
    }
    printf("Inform the list of integers space separated:\n");
    for (int i = 0; i < sb; i++) {
        scanf("%d", &mb[i]);
    }

    printf("Intersection between first and second list:\n");
    for (int i = 0; i < sa; i++) {
        for (int j = 0; j < sb; j++) {
            if (ma[i] == mb[j]) {
                printf("%d ", ma[i]);
                break;
            }
        }
    }
    printf("\n");

    return 0;
}
